### PR TITLE
google-play-ads: show on game pages, show 1st, 5th, 10(n)th visit.

### PR DESCRIPTION
**Pull request by austinhallock**
also updated drawer and gameplayer to use new state stuff

### CODE REVIEW

File: src/components/game_player/index.coffee [hunk 1]
```diff
@@ -20,54 +23,54 @@ module.exports = class GamePlayer
...
+      engagedPlayTimeout:
+        window.setTimeout =>
+          @logEngagedPlay().catch(log.trace)
+        , ENGAGED_PLAY_TIME
 
     User.convertExperiment('game_play').catch log.trace
 
+    if GooglePlayAdService.shouldShowAds()
+      User.getVisitCount().then (visitCount) =>
+        if visitCount is 1 or visitCount is 5 or not visitCount % 10
```
Comment (Zolmeister) [inline src/components/game_player/index.coffee]: magic

File: src/components/game_player/index.coffee [hunk 2]
```diff
@@ -20,54 +23,54 @@ module.exports = class GamePlayer
...
+    @state.set
+      engagedPlayTimeout:
+        window.setTimeout =>
+          @logEngagedPlay().catch(log.trace)
+        , ENGAGED_PLAY_TIME
 
     User.convertExperiment('game_play').catch log.trace
 
+    if GooglePlayAdService.shouldShowAds()
+      User.getVisitCount().then (visitCount) =>
```
Comment (Zolmeister) [inline src/components/game_player/index.coffee]: errors

File: src/components/game_player/index.coffee [hunk 3]
```diff
@@ -20,54 +23,54 @@ module.exports = class GamePlayer
...
+        if visitCount is 1 or visitCount is 5 or not visitCount % 10
+          @showGooglePlayAdModal()
+
+    @showShareModalPromise = Promise.all [
```
Comment (Zolmeister) [inline src/components/game_player/index.coffee]: errors

File: src/components/game_player/index.coffee [hunk 4]
```diff
@@ -20,54 +23,54 @@ module.exports = class GamePlayer
...
+      Game.incrementPlayCount(@state().gameKey)
+      @state.game
+    ]
+    .then ([playCount, game]) =>
+      KikService.isFromPush().then (isFromPush) ->
+        unless isFromPush
+          ga? 'send', 'event', 'game_wo_push', 'game_play', game.key
+      ga? 'send', 'event', 'game', 'game_play', game.key
+      shouldShowModal = playCount is 3
```
Comment (Zolmeister) [inline src/components/game_player/index.coffee]: magic

File: src/components/game_player/index.coffee [hunk 5]
```diff
@@ -20,54 +23,54 @@ module.exports = class GamePlayer
...
+      User.getVisitCount().then (visitCount) =>
+        if visitCount is 1 or visitCount is 5 or not visitCount % 10
+          @showGooglePlayAdModal()
+
+    @showShareModalPromise = Promise.all [
+      Game.incrementPlayCount(@state().gameKey)
+      @state.game
+    ]
+    .then ([playCount, game]) =>
+      KikService.isFromPush().then (isFromPush) ->
```
Comment (Zolmeister) [inline src/components/game_player/index.coffee]: errors

File: src/components/game_player/index.coffee [hunk 6]
```diff
@@ -20,54 +23,54 @@ module.exports = class GamePlayer
   constructor: ({gameKey}) ->
     styles.use()
 
-    @Spinner = new Spinner()
+    o_game = z.observe Game.findOne(key: gameKey)
```
Comment (Zolmeister) [inline src/components/game_player/index.coffee]: rename to `Game.getByKey`

File: src/components/game_player/index.coffee [hunk 7]
```diff
@@ -75,17 +78,32 @@ module.exports = class GamePlayer
...
+      @state.game
+    ]
+    .then ([isFromPush, game]) ->
       unless isFromPush
-        ga? 'send', 'event', 'game_wo_push', 'engaged_play', @game.key
-    ga? 'send', 'event', 'game', 'engaged_play', @game.key
-    User.addRecentGame(@game.id).catch log.trace
+        ga? 'send', 'event', 'game_wo_push', 'engaged_play', game.key
+      ga? 'send', 'event', 'game', 'engaged_play', game.key
+      User.addRecentGame(game.id).catch log.trace
```
Comment (Zolmeister) [inline src/components/game_player/index.coffee]: don't catch here

File: src/components/google_play_ad_modal/index.coffee [hunk 8]
```diff
@@ -10,28 +10,29 @@ module.exports = class GooglePlayAdModal
   constructor: ->
     styles.use()
 
-    @ModalHeader = new ModalHeader {title: 'Get the app'}
+    @ModalHeader = new ModalHeader(
```
Comment (Zolmeister) [inline src/components/google_play_ad_modal/index.coffee]: state

File: src/components/modal_header/index.coffee [hunk 9]
```diff
@@ -15,7 +15,7 @@ module.exports = class ModalHeader
 
   render: =>
     z "div.z-modal-header#{if @isDark then '.is-dark' else ''}",
-      styles:
+      style:
         backgroundImage: if @backgroundImage then "url(#{@backgroundImage})" \
```
Comment (Zolmeister) [inline src/components/modal_header/index.coffee]: state

File: src/models/user.coffee [hunk 10]
```diff
@@ -2,9 +2,11 @@ log = require 'clay-loglevel'
 _ = require 'lodash'
 
 request = require '../lib/request'
+localstore = require '../lib/localstore'
 config = require '../config'
 
 PATH = config.CLAY_API_URL + '/users'
+LOCALSTORE_VISIT_COUNT_KEY = 'User:visitCount'
```
Comment (Zolmeister) [inline src/models/user.coffee]: Follow redis-key style guide: https://github.com/claydotio/wiki/wiki/Ops#redis-keys
(snake_case)

File: src/models/user.coffee [hunk 11]
```diff
@@ -104,6 +106,21 @@ class User
...
           [ op: 'add', path: '/-', value: gameId ]
 
+  incrementVisitCount: ->
+    localstore.get LOCALSTORE_VISIT_COUNT_KEY
+    .then (visitCountObject) ->
+      visitCount = if visitCountObject?.count then visitCountObject.count else 0
+      localstore.set LOCALSTORE_VISIT_COUNT_KEY, {count: visitCount + 1}
+    .then (visitCountObject) ->
+      newVisitCount = visitCountObject.count
+      ga? 'set', 'dimension2', newVisitCount
```
Comment (Zolmeister) [inline src/models/user.coffee]: Where did dimension2 come from? Can it be named better?

File: src/pages/games/index.coffee [hunk 12]
```diff
@@ -34,29 +30,19 @@ module.exports = class GamesPage
       z.redraw()
     .catch log.trace
 
-    User.getExperiments().then (params) =>
-      if params.googlePlayAd isnt 'none' and GooglePlayAdService.shouldShowAds()
-        @GooglePlayAd = if params.googlePlayAd is 'install-button' \
-                        then new GooglePlayAdInstallButton()
-                        else new GooglePlayAdControl()
+    if GooglePlayAdService.shouldShowAds()
+      @GooglePlayAd = new GooglePlayAd()
```
Comment (Zolmeister) [inline src/pages/games/index.coffee]: state

File: src/pages/games/index.coffee [hunk 13]
```diff
@@ -34,29 +30,19 @@ module.exports = class GamesPage
...
+    Modal.openComponent(
+      component: GooglePlayAdModalComponent
+    )
 
   onMount: =>
     if GooglePlayAdService.shouldShowAds()
-      localstore.get('hasSeenGooglePlayAd').then (hasSeenAd) =>
-        unless hasSeenAd
+      User.getVisitCount().then (visitCount) =>
+        if visitCount is 1 or visitCount is 5 or not (visitCount % 10)
```
Comment (Zolmeister) [inline src/pages/games/index.coffee]: duplicated, also magic

File: test/unit/pages/games.coffee [hunk 14]
```diff
@@ -6,28 +6,37 @@ GamesPage = rewire 'pages/games'
 Modal = require 'models/modal'
 
 describe 'GamesPage', ->
-  it 'shows google play modal only on 1st visit', ->
+  it 'shows google play modal on 1st visit', ->
```
Comment (Zolmeister) [inline test/unit/pages/games.coffee]: 5th? 10th?

File: src/components/game_player/index.coffee (hunk 5 above)
Comment (austinhallock) [inline src/components/game_player/index.coffee]: this gets caught below (is part of showShareModalPromise)

Comment (Zolmeister) [inline src/components/game_player/index.coffee]: no it doesn't

File: src/models/user.coffee (hunk 11 above)
Comment (austinhallock) [inline src/models/user.coffee]: Nope.  Will comment

File: src/components/game_player/index.coffee (hunk 2 above)
Comment (austinhallock) [inline src/components/game_player/index.coffee]: done

File: src/components/game_player/index.coffee (hunk 5 above)
Comment (austinhallock) [inline src/components/game_player/index.coffee]: done

**Comment by austinhallock** [comment]
updated

**Comment by Zolmeister** [comment]
fonr

**Comment by austinhallock** [comment]
https://github.com/claydotio/clay-mobile/commit/50231c4270d5b7d5fd1a2c37cba778fcf13e0bc7

**Comment by Zolmeister** [comment]
done

**Comment by austinhallock** [comment]
https://github.com/claydotio/clay-mobile/commit/fae6f5893759da7a1e44abb254a8b4383d5b1288 doesn't include nth thing

**Comment by austinhallock** [comment]
test nth visits https://github.com/claydotio/clay-mobile/commit/96223ac9cbc1f2145f70116605b215e31ac76ea4 (done)

**Comment by Zolmeister** [comment]
squash

**Comment by austinhallock** [comment]
done

**Pull request merged**